tests/conftest.py

```diff
@@ -33,16 +54,89 @@
 }
 
 @pytest.fixture
-def lockfile_with_matching_deployments(tmpdir):
+def lockfile_with_no_deployments(tmpdir):
+    lockfile = copy.deepcopy(LOCKFILE)
+    lockfile.pop("deployments")
     f = tmpdir.join("lockfile.json")
-    f.write(json.dumps(LOCKFILE))
+    f.write(json.dumps(lockfile))
     return str(f)
 
 
 @pytest.fixture
-def lockfile_with_no_deployments(tmpdir):
+def lockfile_with_empty_deployments(tmpdir):
     lockfile = copy.deepcopy(LOCKFILE)
-    lockfile.pop("deployments")
+    lockfile["deployments"] = {}
+    f = tmpdir.join("lockfile.json")
+    f.write(json.dumps(lockfile))
+    return str(f)
+
+
+@pytest.fixture
+def lockfile_with_matching_deployment(w3, tmpdir):
+    w3.testing.mine(5)
+    chain_id = get_chain_id(w3)
+    block = w3.eth.getBlock("earliest")
+    block_uri = create_block_uri(w3.toHex(chain_id), w3.toHex(block.hash))
+    lockfile = copy.deepcopy(LOCKFILE)
+    lockfile["deployments"][block_uri] = {
+      "SafeMathLib": {
+        "contract_type": "SafeMathLib",
+        "address": "0x8d2c532d7d211816a2807a411f947b211569b68c",
+        "transaction": "0xaceef751507a79c2dee6aa0e9d8f759aa24aab081f6dcf6835d792770541cb2b",
+        "block": "0x420cb2b2bd634ef42f9082e1ee87a8d4aeeaf506ea5cdeddaa8ff7cbf911810c"
+      }
+    }
+    f = tmpdir.join("lockfile.json")
+    f.write(json.dumps(lockfile))
+    return str(f)
+
+
+@pytest.fixture
+def lockfile_with_no_matching_deployments(w3, tmpdir):
+    w3.testing.mine(5)
+    incorrect_chain_id = get_chain_id(w3)[:-1] + b'X'
```

Comment:
It'd be better to just use a totally bogus hash rather than modifying the existing one.

```python
incorrect_chain_id = encode_hex(b'\x00' * 31 + b'\x01')  # or just use the longhand string and drop the `encode_hex` call.
```